Compilation error on latest sources

### What happens?

Hello again everyone! Just a compilation error from gcc 11.3.0 from the latest sources I got in the json extension:

```
In file included from compile/extension/json/json_functions/ub_duckdb_json_functions.cpp:10:
src/extension/json/json_functions/json_create.cpp: In member function ‘virtual duckdb::unique_ptr<duckdb::BoundCastData> duckdb::NestedToJSONCastData::Copy() const’:
src/extension/json/json_functions/json_create.cpp:642:24: error: could not convert ‘result’ from ‘unique_ptr<duckdb::NestedToJSONCastData,default_delete<duckdb::NestedToJSONCastData>>’ to ‘unique_ptr<duckdb::BoundCastData,default_delete<duckdb::BoundCastData>>’
  642 |                 return result;
      |                        ^~~~~~
      |                        |
      |                        unique_ptr<duckdb::NestedToJSONCastData,default_delete<duckdb::NestedToJSONCastData>>
```

### To Reproduce

Compile the json extension (-DBUILD_JSON_EXTENSION=ON) with gcc 11.3.0 on debug build type.

### OS:

Linux

### DuckDB Version:

latest from sources

### DuckDB Client:

None

### Full Name:

Pedro Ferreira

### Affiliation:

Huawei

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

--- a/extension/json/json_functions/json_create.cpp
+++ b/extension/json/json_functions/json_create.cpp
@@ -639,7 +639,7 @@ public:
 		for (auto &csn : const_struct_names) {
 			result->const_struct_names.emplace(csn.first, make_uniq<Vector>(csn.second->GetValue(0)));
 		}
-		return result;
+		return std::move(result);
 	}
 
 public:

